### PR TITLE
Implementation of String.prototype.padStart/padEnd.

### PR DESCRIPTION
**Pull request by akroshg**
Implementation of String.prototype.padStart/padEnd.

Proposal for this ES7 feature is found at https://github.com/tc39/proposal-string-pad-start-end.
As the name suggests padStart will add padding to retrieve string up to max length by adding when needed on the front side. The padEnd will add at the rear side.
This feature is sitting behind ES7Builtins flag.
Added tests.

### CODE REVIEW

**Comment by akroshg** [comment]
This was reviewed and signed off earlier - but did not get pushed to master. Rebased commits and created new PR.
@abchatra could you review this? Thanks,

**Comment by abchatra** [comment]
:+1:

File: lib/Runtime/Library/JavascriptString.cpp [hunk 1]
```diff
@@ -1908,6 +1908,89 @@ namespace Js
...
+        }
+
+        if (fillerString == nullptr)
+        {
+            fillerString = NewWithBuffer(L" ", 1, scriptContext);
+        }
+
+        Assert(fillerString->GetLength() > 0);
+
+        charcount_t fillLength = (charcount_t)(maxLength - currentLength); // Possible truncation.
```
Comment (Cellule) [inline lib/Runtime/Library/JavascriptString.cpp]: Is this behavior intended ?
For instance the following code
```js
var s = "1";
print(s.padStart(4294967297));
```
would print `1`

Comment (akroshg) [inline lib/Runtime/Library/JavascriptString.cpp]: @Cellule  Our string does not go more than maxcharcount. 
so in this case there could be one of the two possibilities. 
a) Throw OOM or b) wrap around.
I chose to do wrap around earlier, but now that you pointed that out I looked at other API (eg. repeat) and it throws in such cases. 
I going to match with that and will throw on overflow condition.

Comment (ljharb) [inline lib/Runtime/Library/JavascriptString.cpp]: throwing is preferred, if a correct result won't be returned :+1:

**Comment by ljharb** [comment]
(ﾉ◕ヮ◕)ﾉ*:・ﾟ✧ woot!

When will this be available in a shipping Edge browser? :-D

**Comment by abchatra** [comment]
Changes that make it into our ChakraCore GitHub master branch have a short journey to Chakra.dll. Code flows daily from GitHub to the internal repository from which builds of Chakra.dll are produced and then it flows into Windows and Microsoft Edge. While code flows quickly on this first leg of the journey, code flow from our internal branch to a Windows flighting branch is subject to any number of delays. So it is difficult to predict when your change in our GitHub repo will make it into a particular Windows flight.

In my experience, it takes anywhere between 6 to 10 weeks for the changes to make it to windows technical preview.

**Comment by ljharb** [comment]
Understood, thanks! Where's the best place to keep up with it? It would be amazing, for example, if Edge release announcements included a link to the SHA of ChakraCore that is included :-)

**Comment by abchatra** [comment]
You can following the change log for the technical preview here https://dev.windows.com/en-us/microsoft-edge/platform/changelog/

[Example](https://dev.windows.com/en-us/microsoft-edge/platform/changelog/desktop/10547/)

**Pull request merged**